docs/cody/model-configuration/examples.mdx

```diff
@@ -0,0 +1,681 @@
+          }
+      }
+  ],
+  "modelOverrides": [
+      {
+          "modelRef": "google::unknown::claude-3-5-sonnet",
+          "displayName": "Claude 3.5 Sonnet (via Google/Vertex)",
+          "modelName": "claude-3-5-sonnet@20240620",
+          "contextWindow": {
+            "maxInputTokens": 45000,
+            "maxOutputTokens": 4000
+          },
+          "capabilities": ["chat"],
+          "category": "accuracy",
+          "status": "stable"
+      },
+      {
+          "modelRef": "google::unknown::claude-3-haiku",
+          "displayName": "Claude 3 Haiku",
+          "modelName": "claude-3-haiku@20240307",
+          "capabilities": ["autocomplete", "chat"],
+          "category": "speed",
+          "status": "stable",
+          "tier": "free",
+          "contextWindow": {
+            "maxInputTokens": 7000,
+            "maxOutputTokens": 4000
+          }
+      },
+  ],
+  "defaultModels": {
+    "chat": "google::unknown::claude-3-5-sonnet",
+    "fastChat": "google::unknown::claude-3-5-sonnet",
+    "autocomplete": "google::unknown::claude-3-haiku"
+  }
+}
+```
+
+In the configuration above, we:
+
+-   Set up a provider override for Google Anthropic, routing requests for this provider directly to the specified endpoint (bypassing Cody Gateway).
+-   Add two Anthropic models:
+    -   `"google::unknown::claude-3-5-sonnet"` with "chat" capabiity - used for "chat" and "fastChat"
+    -   `"google::unknown::claude-3-haiku"` with "autocomplete" capability - used for "autocomplete".
+
+</Accordion>
+
+<Accordion title="Google Vertex (Gemini)">
+
+```json
+"modelConfiguration": {
+  "sourcegraph": null,
+  "providerOverrides": [
+      {
+          "id": "google",
+          "displayName": "Google Gemini",
+          "serverSideConfig": {
+            "type": "google",
+            "accessToken": "token",
+            "endpoint": "https://generativelanguage.googleapis.com/v1beta/models"
+          }
+      }
+  ],
+  "modelOverrides": [
+    {
+      "modelRef": "google::v1::gemini-1.5-pro",
+      "displayName": "Gemini 1.5 Pro",
+      "modelName": "gemini-1.5-pro",
+      "capabilities": ["chat", "autocomplete"],
+      "category": "balanced",
+      "status": "stable",
+      "tier": "free",
+      "contextWindow": {
+        "maxInputTokens": 45000,
+        "maxOutputTokens": 400
+      }
+    }
+  ],
+  "defaultModels": {
+    "chat": "google::v1::gemini-1.5-pro",
+    "fastChat": "google::v1::gemini-1.5-pro",
+    "autocomplete": "google::v1::gemini-1.5-pro"
+  }
+}
+```
+
+In the configuration above, we:
+
+-   Set up a provider override for Google Gemini, routing requests for this provider directly to the specified endpoint (bypassing Cody Gateway).
+-   Add the `"google::v1::gemini-1.5-pro"` model, which is used for all Cody features.
+    We do not add other models for simplicity, as adding multiple models is already covered in the examples above.
+
+</Accordion>
+
+<Accordion title="Google Vertex (public)">
```

Comment:
@slimsag, do you know if we have a deployment available to test this setup?

Reply:
I believe Beatrix and Ara both tested this in the past, you may be able to get details from them. Otherwise I think the self-hosted-models GCP project or sourcegraph-dev GCP project might have info for it

Reply:
@aramaraju, do you mean [this](https://console.cloud.google.com/vertex-ai/studio/freeform?model=gemini-1.5-pro-002&inv=1&invt=AbicVA&project=chris-warwick-dev) deployment of Gemini 1.5 Pro? For some reason, the token from 1Pass (Google Vertex section) doesn't work for it. If this deployment is correct, its configuration closely resembles that of Google Vertex (Anthropic), providing a partial example for the use case.
https://github.com/sourcegraph/docs/blob/1033347084a56f5cab1df96e54dded1b22cf431d/docs/cody/enterprise/model-config-examples.mdx#L508-L565

Reply:
@abeatrix Can you share the Gemini 1.5 Pro instructions? I thought it goes to the Gemini API instead of the Vertex API. Might be good to confirm before we push these docs

Reply:
The old instructions are in this PR https://github.com/sourcegraph/sourcegraph-public-snapshot/pull/63053 if users want to go to the Gemini API route instead of Vertex API: https://github.com/sourcegraph/sourcegraph-public-snapshot/pull/63053

This is before we moved to the new model config though:

```json
    "accessToken": "REDACTED",
    "chatModel": "gemini-1.5-pro-latest",
    "provider": "google",
```

@arafatkatze might be a better person to ask for the configuration for going through the Vertex API

Reply:
Hmm.. @taras-yemets let's not block on this, we should go forth with the GCP Vertex Anthropic example, let's remove any Gemini 1.5 Pro examples until we test and verify :).

Reply:
(And thanks, @abeatrix ! We will let the Prime team figure this out post the 5.10 release)